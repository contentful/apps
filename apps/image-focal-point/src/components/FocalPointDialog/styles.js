import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';

export const MAX_PREVIEW_WRAPPER_SIZE = 350;

export const styles = {
  modalContent: css({
    display: 'flex',
  }),
  previewWrapper: css({
    backgroundColor: tokens.gray200,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    position: 'relative',
    width: `${MAX_PREVIEW_WRAPPER_SIZE}px`,
    height: `${MAX_PREVIEW_WRAPPER_SIZE}px`,
  }),
  previewImg: css({
    '&, & > img': {
      cursor: 'crosshair',
      display: 'block',
      margin: '0 auto',
      maxWidth: '100%',
      maxHeight: '100%',
      outline: 0,
    },
    '& > img[src$=".svg"]': {
      width: '100%',
      height: '100%',
      objectFit: 'contain',
    },
  }),
  focalPointDemo: css({
    display: 'flex',
    flexDirection: 'column',
    marginLeft: tokens.spacing2Xl,
  }),
  displayFlex: css({
    display: 'flex',
  }),
  spacingLeftXs: css({
    marginLeft: tokens.spacingXs,
  }),
  previewHeading: css({
    marginTop: tokens.spacingM,
  }),
  subheading: css({
    marginBottom: tokens.spacingXs,
  }),
};
