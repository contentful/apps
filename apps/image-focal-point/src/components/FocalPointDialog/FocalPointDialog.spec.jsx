import React from 'react';
import { render } from '@testing-library/react';

import mockProps from '../../test/mockProps';
import { FocalPointDialog } from './FocalPointDialog';
import { vi } from 'vitest';

const props = {
  file: mockProps.file,
  focalPoint: mockProps.focalPoint,
  onClose: vi.fn(),
  onSave: vi.fn(),
  sdk: mockProps.sdk,
};

describe('FocalPointDialog', () => {
  it('should render the focal point dialog', () => {
    const { container } = render(<FocalPointDialog {...props} />);
    expect(container).toMatchSnapshot();
  });
});
