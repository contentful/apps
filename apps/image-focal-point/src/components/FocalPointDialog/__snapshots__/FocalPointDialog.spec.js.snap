// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocalPointDialog should render the focal point dialog 1`] = `
<div>
  <div
    class="ModalHeader__ModalHeader___1yD1S"
    data-test-id="cf-ui-modal-header"
  >
    <h1
      class="ModalHeader__ModalHeader__title___3IuOy"
    >
      Set focal point
    </h1>
    <button
      class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
      data-test-id="cf-ui-icon-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
        tabindex="-1"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--primary___3M4Il IconButton__IconButton__icon___3yZQN"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
          />
        </svg>
        <span
          class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
        >
          Close
        </span>
      </span>
    </button>
  </div>
  <div
    class="ModalContent__ModalContent___2mf3h"
    data-test-id="cf-ui-modal-content"
  >
    <div
      class="css-k008qs"
    >
      <div>
        <h2
          class="Subheading__Subheading___2mA9j css-cet0rr"
          data-test-id="cf-ui-subheading"
        >
          Select position of focal point
        </h2>
        <div
          class="css-1biw769"
        >
          <div
            class="css-1njnee2"
            role="button"
            tabindex="-1"
          >
            <img
              alt="focal point preview"
              src="https://mycdn.com/some_image.png"
            />
          </div>
        </div>
      </div>
      <div
        class="css-1p7lqom"
      >
        <h2
          class="Subheading__Subheading___2mA9j css-cet0rr"
          data-test-id="cf-ui-subheading"
        >
          Preview for different screen sizes
        </h2>
        <div
          class="css-k008qs"
        >
          <div
            class=""
          >
            <div
              style="width: 410px; height: 180px; overflow: hidden; position: relative;"
            >
              <img
                alt="Preview with cropping applied"
                src="https://mycdn.com/some_image.png"
                style="width: 410px; height: 276px; transform: translate3d(0px, -28.72524590163934px, 0); transition: transform 0.3s cubic-bezier(0.13, 0.62, 0.11, 0.99);"
              />
            </div>
            <p
              class="css-qslnu8"
            >
              Desktop
            </p>
          </div>
        </div>
        <div
          class="css-k008qs"
        >
          <div
            class=""
          >
            <div
              style="width: 280px; height: 180px; overflow: hidden; position: relative;"
            >
              <img
                alt="Preview with cropping applied"
                src="https://mycdn.com/some_image.png"
                style="width: 280px; height: 188px; transform: translate3d(0px, 0px, 0); transition: transform 0.3s cubic-bezier(0.13, 0.62, 0.11, 0.99);"
              />
            </div>
            <p
              class="css-qslnu8"
            >
              Tablet
            </p>
          </div>
          <div
            class="css-1f2k2gl"
          >
            <div
              style="width: 120px; height: 180px; overflow: hidden; position: relative;"
            >
              <img
                alt="Preview with cropping applied"
                src="https://mycdn.com/some_image.png"
                style="width: 268px; height: 180px; transform: translate3d(0px, 0px, 0); transition: transform 0.3s cubic-bezier(0.13, 0.62, 0.11, 0.99);"
              />
            </div>
            <p
              class="css-qslnu8"
            >
              Mobile
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ModalControls__ModalControls___2bTQx"
    data-test-id="cf-ui-modal-controls"
  >
    <button
      class="Button__Button___1ZfFj Button__Button--positive___1t6w1"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Save
        </span>
      </span>
    </button>
    <button
      class="Button__Button___1ZfFj Button__Button--muted___2Wair"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Close
        </span>
      </span>
    </button>
  </div>
</div>
`;
